UnitTests/GitUITests/CommandsDialogs/CommitDialog/FormCommitTests.cs

```diff
@@ -86,64 +85,49 @@ public void Should_update_committer_info_on_form_activated()
         [Test]
         public void Should_display_branch_and_no_remote_info_in_statusbar()
         {
-            using (var repository = new ReferenceRepository(new GitModuleTestHelper("branch_and_no_remote")))
+            _referenceRepository.CheckoutMaster();
+            RunFormTest(form =>
             {
-                var commands = new GitUICommands(repository.Module);
-                repository.CheckoutMaster();
-                UITest.RunForm<FormCommit>(
-                    () => { Assert.True(commands.StartCommitDialog(owner: null)); },
-                    form =>
-                    {
-                        var currentBranchNameLabelStatus = form.GetTestAccessor().CurrentBranchNameLabelStatus;
-                        var remoteNameLabelStatus = form.GetTestAccessor().RemoteNameLabelStatus;
+                form.GetTestAccessor().RunningTasks.JoinTillEmptyAsync();
 
-                        Assert.AreEqual("master →", currentBranchNameLabelStatus.Text);
-                        Assert.AreEqual("(remote not configured)", remoteNameLabelStatus.Text);
-                        return Task.CompletedTask;
-                    });
-            }
+                var currentBranchNameLabelStatus = form.GetTestAccessor().CurrentBranchNameLabelStatus;
+                var remoteNameLabelStatus = form.GetTestAccessor().RemoteNameLabelStatus;
+
+                Assert.AreEqual("master →", currentBranchNameLabelStatus.Text);
+                Assert.AreEqual("(remote not configured)", remoteNameLabelStatus.Text);
+            });
         }
 
         [Test]
         public void Should_display_detached_head_info_in_statusbar()
         {
-            using (var repository = new ReferenceRepository(new GitModuleTestHelper("detached_head")))
+            _referenceRepository.CheckoutRevision();
+            RunFormTest(form =>
             {
-                var commands = new GitUICommands(repository.Module);
-                repository.CheckoutRevision();
-                UITest.RunForm<FormCommit>(
-                    () => { Assert.True(commands.StartCommitDialog(owner: null)); },
-                    form =>
-                    {
-                        var currentBranchNameLabelStatus = form.GetTestAccessor().CurrentBranchNameLabelStatus;
-                        var remoteNameLabelStatus = form.GetTestAccessor().RemoteNameLabelStatus;
+                form.GetTestAccessor().RunningTasks.JoinTillEmptyAsync();
```

Comment:
❗️ This needs to use something like this:

```suggestion
                ThreadHelper.JoinableTaskFactory.Run(() => form.GetTestAccessor().RunningTasks.JoinTillEmptyAsync());
```